compiler/src/iree/compiler/Dialect/Stream/Transforms/EncodeTensors.cpp

```diff
@@ -346,7 +306,22 @@ static Value canonicalizeFillPattern(Value pattern, PatternRewriter &rewriter) {
   if (patternType.isInteger(1)) {
     return rewriter.createOrFold<arith::ExtUIOp>(loc, rewriter.getI8Type(),
                                                  pattern);
-  } else if ((bitWidth % 8) != 0) {
+  }
+  // For packed sub-byte patterns, duplicate the sub-byte parts into a full
+  // byte.
+  if (needToPackSubByteInterfaceBitWidth(bitWidth)) {
+    Type i8Type = rewriter.getI8Type();
+    Value bw = rewriter.createOrFold<arith::ConstantOp>(
+        loc, i8Type, rewriter.getIntegerAttr(i8Type, bitWidth));
+    Value part = rewriter.createOrFold<arith::ExtUIOp>(loc, i8Type, pattern);
+    Value full = part;
+    for (unsigned i = 1, e = 8 / bitWidth; i < e; ++i) {
+      Value shifted = rewriter.createOrFold<arith::ShLIOp>(loc, full, bw);
+      full = rewriter.createOrFold<arith::OrIOp>(loc, shifted, part);
+    }
+    return full;
+  }
```

Comment:
I can't quite read what this is doing - can you put the expression in a comment/use more descriptive variable names?

If this is doing alignment (round up to power of two) you can use `util.align`, which we have some analysis for.

Reply:
This is duplicating the sub-byte value into a full-byte. I've updated the comment to make it clear.